src/webgpu/api/validation/encoding/cmds/render/setVertexBuffer.spec.ts

```diff
@@ -0,0 +1,108 @@
+export const description = `
+Validation tests for setVertexBuffer on render pass and render bundle.
+`;
+
+import { makeTestGroup } from '../../../../../../common/framework/test_group.js';
+import { GPUConst, DefaultLimits } from '../../../../../constants.js';
+import { ValidationTest, kRenderEncodeTypes } from '../../../validation_test.js';
+
+export const g = makeTestGroup(ValidationTest);
+
+g.test('slot')
+  .desc(
+    `
+Tests slot must be less than the maxVertexBuffers in device limits.
+  `
+  )
+  .paramsSubcasesOnly(u =>
+    u
+      .combine('encoder', kRenderEncodeTypes)
+      .combine('slot', [
+        0,
+        DefaultLimits.maxVertexBuffers - 1,
+        DefaultLimits.maxVertexBuffers,
+      ] as const)
+  )
+  .unimplemented();
+
+g.test('vertex_buffer')
+  .desc(
+    `
+Tests vertex buffer must be valid.
+  `
+  )
+  .paramsSubcasesOnly(u =>
+    u
+      .combine('encoder', kRenderEncodeTypes)
+      .combine('state', ['valid', 'invalid', 'destroyed'] as const)
+  )
+  .unimplemented();
+
+g.test('vertex_buffer_usage')
+  .desc(
+    `
+Tests vertex buffer must have 'Vertex' usage.
+  `
+  )
+  .paramsSubcasesOnly(u =>
+    u.combine('encoder', kRenderEncodeTypes).combine('usage', [
+      GPUConst.BufferUsage.VERTEX, // control case
+      GPUConst.BufferUsage.COPY_DST,
+      GPUConst.BufferUsage.COPY_DST | GPUConst.BufferUsage.VERTEX,
+    ] as const)
+  )
+  .unimplemented();
+
+g.test('offset_alignment')
+  .desc(
+    `
+Tests offset must be a multiple of 4.
+  `
+  )
+  .paramsSubcasesOnly(u =>
+    u.combine('encoder', kRenderEncodeTypes).combine('offset', [0, 2, 4] as const)
+  )
+  .unimplemented();
+
+g.test('offset_and_size_oob')
+  .desc(
+    `
+Tests offset and size cannot be larger than vertex buffer size
+- (offset, size) is
+  - (0, 1)
+  - (0, min size) (control case)
```

Comment:
The smallest vertex format is only 2 bytes, and could be 1 byte in the future. I don't think there are any restrictions on min size of a vertex buffer binding, at least I don't see them in the spec right now.

Reply:
Here I want the 'min size' to be the min size of an index / vertex buffer, depends on the data type in the buffer. The OOB validation has nothing to do with the data type, I thought is to use uint32 for index and float32 for vertex to have a same 'min size'.

Reply:
A particular pipeline may need a certain size for each bound vertex and index buffer, but I don't see any restrictions on a setVertexBuffer call by itself. Both setIndexBuffer and setVertexBuffer do have alignment restrictions on offset, but not on size (at least in the current spec draft).

Reply:
Yes. The 'min size' does not come from SPEC. I saw it at [1]. If I misunderstood the 'min size', what should it be?

Reply:
Ah, sorry I didn't find the existing notes that said "min size". I think it means means "whatever is the minimum setIndexBuffer `size` argument value (maybe it's 1, wasn't sure when writing this)". The point is to test the case that's just barely out of bounds, to make sure there are no off-by-one errors.

I think it's kind of redundant, e.g. `(0, 1)` and `(0, min size)` are probably the same thing.